gluefactory/models/lines/lsd.py

```diff
@@ -55,32 +51,37 @@ def detect_lines(self, img):
     def _forward(self, data):
         # Convert to the right data format
         image = data["image"]
+
         if image.shape[1] == 3:
             # Convert to grayscale
             scale = image.new_tensor([0.299, 0.587, 0.114]).view(1, 3, 1, 1)
             image = (image * scale).sum(1, keepdim=True)
+
         device = image.device
         b_size = len(image)
         image = np.uint8(image.squeeze(1).cpu().numpy() * 255)
 
-        # LSD detection in parallel
+        segs = batched_lsd(image)
+
         if b_size == 1:
```

Comment:
We don't need the check if batch_size == 1 here anymore. Simply looping through all segs should also work for b_size = 1.